docs/migration_guide.md

```diff
@@ -40,7 +40,9 @@ Because of confusion with `Options` defaults in bincode 1, we have made `Configu
 
 ## Migrating with `serde`
 
-Make sure to include bincode 2 with the `serde` feature enabled.
+You may wish to stick with `serde` when migrating to bincode 2, for example if you are using serde-exclusive derive features such as `#[serde(deserialize_with)]`.
+
+If so, make sure to include bincode 2 with the `serde` feature enabled.
```

Comment:
Might be useful to also mention that people should use the `bincode::serde::*` functions instead of the `bincode::*` functions